airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -472,3 +528,207 @@ class Labels(GoogleAdsStream):
     """
 
     primary_key = ["label.id"]
+
+
+class ChangeStatus(IncrementalGoogleAdsStream):
+    """
+    Change status stream: https://developers.google.com/google-ads/api/fields/v14/change_status
+    Stream is only used internally to implement incremental updates for child streams of IncrementalEventsStream
+    """
+
+    cursor_field = "change_status.last_change_date_time"
+    slice_step = pendulum.duration(microseconds=1)
+    days_of_data_storage = 90
+    cursor_time_format = "YYYY-MM-DD HH:mm:ss.SSSSSS"
+    mandatory_limit = 10000
+
+    def __init__(self, **kwargs):
+        # date range is not used for these streams, only state is used to sync recent records, otherwise full refresh
+        for key in ["start_date", "conversion_window_days", "end_date"]:
+            kwargs.pop(key, None)
+        super().__init__(start_date=None, conversion_window_days=0, end_date=None, **kwargs)
+
+    def get_query(self, stream_slice: Mapping[str, Any]) -> str:
+        query = GoogleAds.convert_schema_into_query(
+            schema=self.get_json_schema(),
+            report_name=self.name,
+            from_date=stream_slice.get("start_date"),
+            to_date=stream_slice.get("end_date"),
+            cursor_field=self.cursor_field,
+            # resource type is used for filtering updates only for desirable stream
+            resource_type=stream_slice.get("resource_type"),
+            # limit is mandatory parameter for this stream, also 10k is maximum value
+            limit=self.mandatory_limit,
+        )
+        return query
+
+
+class IncrementalEventsStream(GoogleAdsStream, IncrementalMixin, ABC):
+    """
+    Abstract class used for getting incremental updates based on events returned from ChangeStatus stream
+    """
+
+    def __init__(self, parent_stream: IncrementalGoogleAdsStream, **kwargs):
+        self.parent_stream = parent_stream
```

Comment:
why is parent stream parametrized? isn't it always `ChangeStatus`?

Reply:
Fixed